File: Easy/922. Sort Array By Parity II/SortArrayByParityII.java
/*
LeetCode Easy: 922. Sort Array By Parity II
https://leetcode.com/problems/sort-array-by-parity-ii/

Check SortArrayByParityII.cpp for brief solution explanation.
*/
class Solution {
    public int[] sortArrayByParityII(int[] nums) {
        int N = nums.length;
        int i = 0, j = 1;
        while(i<N&&j<N){
            while(i<N&&nums[i]%2==0){
                i+=2;
            }
            while(j<N&&nums[j]%2==1){
                j+=2;
            }
            if(i<N&&j<N){
                int t = nums[i];
                nums[i] = nums[j];
                nums[j] = t;
            }else if(i>=N||j>=N){
                break;
            }
        }
        return nums;
    }
}

File: Easy/922. Sort Array By Parity II/SortArrayByParityII.py
'''
LeetCode Easy: 922. Sort Array By Parity II
https://leetcode.com/problems/sort-array-by-parity-ii/

Check SortArrayByParityII.cpp for brief solution explanation.
'''
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        N = len(nums)
        evens, odds = [], []
        for i in range(N):
            if nums[i]%2==0:
                evens.append(nums[i])
            else:
                odds.append(nums[i])
        ans = []
        p1, p2 = 0, 0
        for i in range(N):
            if i%2==0:
                ans.append(evens[p1])
                p1+=1
            else:
                ans.append(odds[p2])
                p2+=1
        return ans